mempool/registry_production.go

```diff
@@ -0,0 +1,33 @@
+//go:build !test
+// +build !test
+
+package mempool
+
+import "errors"
+
+// globalEVMMempool holds the global reference to the ExperimentalEVMMempool instance.
+// It can only be set during application initialization.
+var globalEVMMempool *ExperimentalEVMMempool
```

Comment:
could the global mempool be an interface so that we can just have different implemenations (prod and test) instead of using build flags?

We don't have to do this now, I just want to get us away from these build flag shenanigans eventually

Reply:
Yeah a good idea for cleanup

Reply:
yeah i will create public task for this